### PR TITLE
Cabal 3.2 support

### PR DESCRIPTION
**Pull request by angerman**
We'll need to support Cabal-3.2 to that hackage.nix properly updated, and we can parse `.cabal` files using new cabal features/formats. This requires to upgrade `nix-tools` to build against the `Cabal-3.2`.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
-      url = "https://github.com/input-output-hk/haskell.nix/archive/a84e3b55c642e7aec57c9677158cf446d90e9759.tar.gz";
-      sha256 = "0qnchf0f8d5k6363mcrqv936wz6zljs06bjvg1i6xmkg31bdj48m";
+      url = "https://github.com/input-output-hk/haskell.nix/archive/f78841c2d05c6c686d3888131104360944464dc2.tar.gz";
+      sha256 = "1zwhskx9drd0266kcrxf5vzcgs9vfiiibzkc2h3yarfb2xl7185q";
     }
-, nixpkgs ? haskellNixSrc + "/nixpkgs"
-, pkgs ? import nixpkgs (import haskellNixSrc)
+, nixpkgs ? (import haskellNixSrc {}).sources.nixpkgs-default
+, pkgs ? import nixpkgs (import haskellNixSrc {}).nixpkgsArgs
 , haskellCompiler ? "ghc865"
```
Comment (michaelpj) [inline default.nix:7]: Should we bump this too?

**Comment by maksbotan** [comment]
I tried to follow buildkite script locally, but it seems to be broken in many ways.

How does it even build? For me it fails, for example, because `saltline` is pulled in and it depends on `sodium` library in `libsodium` package. And there is no mapping `sodium -> libsodium` in haskell.nix.

**Comment by maksbotan** [comment]
I'm using ghc-8.8.3 by the way.

**Comment by angerman** [comment]
@maksbotan I think `cabal new-build` should work in the nix-shell. We may have to fix the buildkite script as well.

File: default.nix (hunk 1 above)
Comment (angerman) [reply default.nix:7]: @michaelpj the compiler you mean?

**Comment by maksbotan** [comment]
Okay, let me try in the shell.
BTW, how can I see buildkite logs? I created an account, but can't open the page anyway.

File: default.nix (hunk 1 above)
Comment (michaelpj) [reply default.nix:7]: Yeah - I don't know how much it matters, though.

**Comment by angerman** [comment]
@maksbotan this might be something that requires a company account. Quite unfortunate. Maybe we should try the nix integration for GitHub actions. I'll have a look.

**Comment by maksbotan** [comment]
@hamishmack @angerman please see https://github.com/maksbotan/nix-tools/tree/maksbotan/cabal-32 (diff with this PR is here: https://github.com/input-output-hk/nix-tools/compare/hkm/cabal-32...maksbotan:maksbotan/cabal-32?expand=1)

It builds fine for me in `nix-shell` with GHC-8.8.3.

Also this will be needed: https://github.com/input-output-hk/haskell.nix/pull/587

If everything is OK, I think you should just merge my branch into yours.

**Pull request merged**